### PR TITLE
fix: override version to release SHA and remove protobuf patch

### LINKED ISSUE #55: Cannot use extensions because build SHA is wrong
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The current 0.4.0 release of `duckdb` has a commit SHA of https://github.com/duckdb/duckdb/commit/da9ee490df829a96bfbcfcd737f95f8dbc707d0a

The new extension installation procedure relies on URL construction based on the short SHA -- I believe that the protobuf patch being applied in the build here is changing the sha and that's what is eventually built (although it could be something else).

With the current 0.4.0 build, extension installation fails:
```
In [4]: con.execute("install substrait")
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 con.execute("install substrait")

RuntimeError: IO Error: Failed to download extension http://extensions.duckdb.org/c1bb9f6/linux_amd64/substrait.duckdb_extension.gz
```

note that `c1bb9f6` does not correspond to any commit in `duckdb`

FWIW, I think the protobuf patch was probably a workaround for the substrait protobuf stuff bundled with previous versions -- it is no longer bundled (and is handled by the new extension system)

### Installed packages

```shell
(duckdb_condaforge_test) gil@stravaig ~ 🐚  conda list
# packages in environment at /home/gil/mambaforge/envs/duckdb_condaforge_test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
abseil-cpp                20210324.2           h9c3ff4c_0    conda-forge
arrow-cpp                 9.0.0           py310hd6ac832_1_cpu    conda-forge
asttokens                 2.0.8              pyhd8ed1ab_0    conda-forge
aws-c-cal                 0.5.11               h95a6274_0    conda-forge
aws-c-common              0.6.2                h7f98852_0    conda-forge
aws-c-event-stream        0.2.7               h3541f99_13    conda-forge
aws-c-io                  0.10.5               hfb6a706_0    conda-forge
aws-checksums             0.1.11               ha31a3da_7    conda-forge
aws-sdk-cpp               1.8.186              hb4091e7_3    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2022.6.15            ha878542_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
executing                 0.10.0             pyhd8ed1ab_0    conda-forge
gflags                    2.2.2             he1b5a44_1004    conda-forge
glog                      0.6.0                h6f12383_0    conda-forge
grpc-cpp                  1.45.2               h9d3bbbb_5    conda-forge
ipython                   8.4.0           py310hff52083_0    conda-forge
jedi                      0.18.1             pyhd8ed1ab_2    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.19.3               h3790be6_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libblas                   3.9.0           16_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h166bdaf_7    conda-forge
libbrotlidec              1.0.9                h166bdaf_7    conda-forge
libbrotlienc              1.0.9                h166bdaf_7    conda-forge
libcblas                  3.9.0           16_linux64_openblas    conda-forge
libcrc32c                 1.1.2                h9c3ff4c_0    conda-forge
libcurl                   7.83.1               h7bff187_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               h9b69904_4    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.1.0              h8d9b700_16    conda-forge
libgfortran-ng            12.1.0              h69a702a_16    conda-forge
libgfortran5              12.1.0              hdcd56e2_16    conda-forge
libgomp                   12.1.0              h8d9b700_16    conda-forge
libgoogle-cloud           1.40.2               habd0e3a_0    conda-forge
liblapack                 3.9.0           16_linux64_openblas    conda-forge
libnghttp2                1.47.0               hdcd2b5c_1    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.21          pthreads_h78a6416_2    conda-forge
libprotobuf               3.20.1               h6239696_1    conda-forge
libsqlite                 3.39.2               h753d276_1    conda-forge
libssh2                   1.10.0               haa6b8db_3    conda-forge
libstdcxx-ng              12.1.0              ha89aaad_16    conda-forge
libthrift                 0.16.0               h519c5ea_1    conda-forge
libutf8proc               2.7.0                h7f98852_0    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.12               h166bdaf_2    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
numpy                     1.23.2          py310h53a5b5f_0    conda-forge
openssl                   1.1.1q               h166bdaf_0    conda-forge
orc                       1.7.5                h6c59b99_0    conda-forge
pandas                    1.4.3           py310h769672d_0    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pip                       22.2.2             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.30             pyha770c72_0    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyarrow                   9.0.0           py310h580ff74_1_cpu    conda-forge
pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
python                    3.10.5          h582c2e5_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-duckdb             0.4.0           py310h21b4e68_0    conda-forge
python_abi                3.10                    2_cp310    conda-forge
pytz                      2022.2.1           pyhd8ed1ab_0    conda-forge
re2                       2022.06.01           h27087fc_0    conda-forge
readline                  8.1.2                h0f457ee_0    conda-forge
s2n                       1.0.10               h9b69904_0    conda-forge
setuptools                65.2.0          py310hff52083_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.9                hbd366e4_1    conda-forge
sqlite                    3.39.2               h4ff8645_1    conda-forge
stack_data                0.4.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
traitlets                 5.3.0              pyhd8ed1ab_0    conda-forge
tzdata                    2022c                h191b570_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.2.12               h166bdaf_2    conda-forge
zstd                      1.5.2                h6239696_4    conda-forge
```

### Environment info

```shell
(duckdb_condaforge_test) gil@stravaig ~ 🐚  conda info

     active environment : duckdb_condaforge_test
    active env location : /home/gil/mambaforge/envs/duckdb_condaforge_test
            shell level : 2
       user config file : /home/gil/.condarc
 populated config files : /home/gil/mambaforge/.condarc
                          /home/gil/.condarc
          conda version : 4.13.0
    conda-build version : not installed
         python version : 3.10.5.final.0
       virtual packages : __linux=5.15.61=0
                          __glibc=2.36=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/gil/mambaforge  (writable)
      conda av data dir : /home/gil/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/gil/mambaforge/pkgs
                          /home/gil/.conda/pkgs
       envs directories : /home/gil/mambaforge/envs
                          /home/gil/.conda/envs
               platform : linux-64
             user-agent : conda/4.13.0 requests/2.27.1 CPython/3.10.5 Linux/5.15.61-1-lts arch/ glibc/2.36
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by gforsyth**
Substrait is no longer bundled with DuckDB so the protobuf patches
shouldn't be required anymore.  This also sets the `PRETEND_VERSION`
override to match the upstream release SHA so that extension URL
resolution works.

Should fix #55

### CODE REVIEW

File: recipe/meta.yaml
Comment (xhochy) [inline recipe/meta.yaml:25]: This looks a bit weird, why isn't this the package version anymore?

Comment (gforsyth) [reply recipe/meta.yaml:25]: Because extensions in duckdb are hosted with a URL scheme like

`http://extensions.duckdb.org/{SHA1}/linux_amd64/{extension_name}.duckdb_extension.gz`

and so if the setuptools version doesn't match the upstream SHA they aren't accessible.

Comment (Mause) [reply recipe/meta.yaml:25]: Pretty sure the hash is stored in `SETUPTOOLS_SCM_PRETEND_HASH`, `SETUPTOOLS_SCM_PRETEND_VERSION` should still be `{{ version }}`, see https://github.com/duckdb/duckdb/blob/5c9a3d68662075ce3d5be9edfe02b524036f4fdb/scripts/package_build.py#L108-L134

**Review by Mause — CHANGES_REQUESTED** [review]

**Comment by dhirschfeld** [comment]
Thought I'd try `SETUPTOOLS_SCM_PRETEND_HASH` but it looks like it's still failing `test_stubs`:
```
=================================== FAILURES ===================================
__________________________________ test_stubs __________________________________

    def test_stubs():
    	# The test has false positives on the following keywords
    	skip_stubs_errors = ['pybind11_module_local', 'git_revision']
    	# just run stubtest
    	stubs = subprocess.run([
    			sys.executable, '-m', 'mypy.stubtest',
    			'duckdb',
    			'--mypy-config-file', MYPY_INI_PATH,
    		], stdout=subprocess.PIPE)
    	if (stubs.returncode != 0):
    		errors = stubs.stdout.decode('utf-8').split('error')
    		broken_stubs = []
    		for error in errors:
    			add_error = True
    			for skip in skip_stubs_errors:
    				if error == '' or skip in error:
    					add_error = False
    					break
    			if (add_error):
    				broken_stubs.append(error)
    			if len(broken_stubs) > 0:
    				print("Stubs must be updated, either add them to skip_stubs_errors or update __init__.pyi accordingly")
    				print(broken_stubs)
>   				assert(0)
E       assert 0

tools/pythonpkg/tests/stubs/test_stubs.py:34: AssertionError
----------------------------- Captured stdout call -----------------------------
Stubs must be updated, either add them to skip_stubs_errors or update __init__.pyi accordingly
['s (checked 1 module)\n']
=========================== short test summary info ============================
FAILED tools/pythonpkg/tests/stubs/test_stubs.py::test_stubs - assert 0
============ 1 failed, 401 passed, 15 xpassed in 158.69s (0:02:38) =============v
```

**Comment by Mause** [comment]
It might be worth running that test locally, as it's possible the parsing of stubtests stdout is failing

**Comment by gforsyth** [comment]
> It might be worth running that test locally, as it's possible the parsing of stubtests stdout is failing

Yeah, looks like some extra output at the end of newer mypy runs is borking this.  I'll submit a PR upstream and add a patch here for now

**Comment by gforsyth** [comment]
@xhochy @dhirschfeld @mariusvniekerk -- this is ready for another look

**Review by dhirschfeld — APPROVED** [review]
If the CI is happy, I'm happy but I'll let one of the other maintainers do the honours as I'm not fully across the details of this build...

**Comment by mariusvniekerk** [comment]
:shipit:

**Comment by dhirschfeld** [comment]
Will merge in the morning if no one objects / beats me to it...

**Pull request merged**
